ElectronClient/gui/NoteContentPropertiesDialog.tsx

```diff
@@ -19,30 +20,68 @@ interface KeyToLabelMap {
 	[key: string]: string;
 }
 
+interface Counter {
+	words: number;
+	all: number;
+	characters: number;
+}
+
 export default function NoteContentPropertiesDialog(props:NoteContentPropertiesDialogProps) {
 	const theme = themeStyle(props.theme);
-	const textComps: JSX.Element[] = [];
+	const tableBodyComps: JSX.Element[] = [];
+	// For the source Markdown
 	const [lines, setLines] = useState<number>(0);
 	const [words, setWords] = useState<number>(0);
 	const [characters, setCharacters] = useState<number>(0);
 	const [charactersNoSpace, setCharactersNoSpace] = useState<number>(0);
+	// For source with Markdown syntax stripped out
+	const [strippedLines, setStrippedLines] = useState<number>(0);
+	const [strippedWords, setStrippedWords] = useState<number>(0);
+	const [strippedCharacters, setStrippedCharacters] = useState<number>(0);
+	const [strippedCharactersNoSpace, setStrippedCharactersNoSpace] = useState<number>(0);
 
 	useEffect(() => {
-		Countable.count(props.text, (counter: { words: number; all: number; characters: number; }) => {
+		Countable.count(props.text, (counter: Counter) => {
 			setWords(counter.words);
 			setCharacters(counter.all);
 			setCharactersNoSpace(counter.characters);
 		});
 		props.text === '' ? setLines(0) : setLines(props.text.split('\n').length);
 	}, [props.text]);
 
+	useEffect(() => {
+		const strippedText: string = removeMd(props.text, {
+			gfm: false,
+		}); // After stripping out Markdown syntax
+		if (strippedText === props.text) {
+			setStrippedWords(words);
+			setStrippedCharacters(characters);
+			setStrippedCharactersNoSpace(charactersNoSpace);
+			setStrippedLines(lines);
+		} else {
+			Countable.count(strippedText, (counter : Counter) => {
+				setStrippedWords(counter.words);
+				setStrippedCharacters(counter.all);
+				setStrippedCharactersNoSpace(counter.characters);
+			});
+			strippedText === '' ? setStrippedLines(0) : setStrippedLines(strippedText.split('\n').length);
+		}
+	}, [props.text]);
```

Comment:
We need unit tests for this:
- Counting words, characters, etc. on Markdown text
- Counting words, characters, etc. on stripped text

Reply:
I am working on this. Basically, we need to ensure that both `countable` and `remove-markdowon` seperately work fine.

Reply:
Maybe you can expect this logic to utility functions so that it's easier to test.

Reply:
@laurent22, I have refactored the code you were referring to. Do I need to test for every sort of Markdown syntax in the unit test or only a specific part? Because there are quite a few Markdown elements.

Reply:
@laurent22 , I submitted the tests for the stripping of markdown.